src/components/DownloadButton.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { set } from 'idb-keyval';
+import GetAppIcon from '@material-ui/icons/GetApp';
+import { videoOrArticle, VideoData, ArticleData } from '../store/resourcesSlice'
+import { videoStore } from '../index'
+import Button from '@material-ui/core/Button';
+import { useDispatch } from 'react-redux';
+import { setResourceIsCached } from '../store/resourcesSlice';
+
+type DownloadButtonProps = {
+  id: number,
+  videoData?: VideoData | ArticleData,
+}
+
+const DownloadButton = ({ id, videoData }: DownloadButtonProps) => {
+  const dispatch = useDispatch();
+
+  async function downloadVideo(resId, videoStore, videoData) {
+    let downloadUrl = videoOrArticle(videoData) ? videoData.downloadUrl : null
+    if (downloadUrl) {
+      let blob = await fetch(downloadUrl).then(res => res.blob())
```

Comment:
Saving blobs to IndexedDB may not be supported on Safari (see [this link](https://developers.google.com/web/fundamentals/instant-and-offline/web-storage/indexeddb-best-practices#not_everything_can_be_stored_in_indexeddb_on_all_platforms)). If this is important to us, we should convert the blob to an ArrayBuffer before saving it (the link explains how to do that) to guarantee that it works across all platforms. The code itself isn't too complicated but there would be some extra overhead when saving and loading videos

@joelenelatief is iOS support important?

Reply:
iOS support is secondary in priority! @susan-lin @kevintli, thoughts on implementing some [detection](https://stackoverflow.com/questions/21741841/detecting-ios-android-operating-system) to see if we're on an iOS or Android device before converting, so that Android doesn't have the extra overhead?